appserver/jdbc/jdbc-config/src/main/java/org/glassfish/jdbc/config/validators/JdbcConnectionPoolValidator.java

```diff
@@ -65,23 +68,75 @@ public void initialize(final JdbcConnectionPoolConstraint constraint) {
         this.poolFaults = constraint.value();
     }
 
+    public String getParsedVariable(String variableToRetrieve) {
+
+        String[] variableReference = variableToRetrieve.split("=");
+        
+        switch(variableReference[0]) {
+            case "ENV":
+                //Check environment variables for requested value
+                if (System.getenv(variableReference[1]) != null && !System.getenv(variableReference[1]).isEmpty()) {
+                    return System.getenv(variableReference[1]);
+                }
+                break;
+            case "MPCONFIG":
+                //Check microprofile config for requested value
+                Config config = ConfigProvider.getConfig();
+                if (config.getValue(variableReference[1], String.class) != null && !config.getValue(variableReference[1], String.class).isEmpty()) {
+                    return config.getValue(variableReference[1], String.class);
+                }
+                break;
+            default:
+                //We got a system variable as no split occured
+                if (System.getProperty(variableReference[0]) != null && !System.getProperty(variableReference[0]).isEmpty()) {
+                    return System.getProperty(variableReference[0]);
+                }
+                break;
+        }
+        
+        //If this point is reached, the variable value could not be found
+        return null; 
+    }
+    
     @Override
     public boolean isValid(final ResourcePool pool,
         final ConstraintValidatorContext constraintValidatorContext) {
 
         if(poolFaults == ConnectionPoolErrorMessages.MAX_STEADY_INVALID) {
             if(pool instanceof JdbcConnectionPool) {
+                
                 JdbcConnectionPool jdbcPool = (JdbcConnectionPool) pool;
                 String maxPoolSize = jdbcPool.getMaxPoolSize();
                 String steadyPoolSize = jdbcPool.getSteadyPoolSize();
-                if(steadyPoolSize == null) {
+                
+                int maxPoolSizeValue = 0;
+                int steadyPoolSizeValue = 0;
```

Comment:
It should be possible to add a helper function where the min/max/default and actual value go in and the successfully parsed or resolved value comes out that can be used for both of these. Essentially making a method that does the extended version of that the annotations did semantically.